python/ray/data/tests/test_streaming_executor_resource_limits.py

```diff
@@ -0,0 +1,102 @@
+import collections
+import math
+import time
+from unittest.mock import patch
+
+import pytest
+
+import ray
+from ray.data._internal.execution.interfaces import ExecutionOptions, ExecutionResources
+from ray.data._internal.execution.operators.map_transformer import (
+    create_map_transformer_from_block_fn,
+)
+from ray.data._internal.execution.streaming_executor import StreamingExecutor
+from ray.data._internal.execution.streaming_executor_state import (
+    DEFAULT_OBJECT_STORE_MEMORY_LIMIT_FRACTION,
+    DownstreamMemoryInfo,
+    TopologyResourceUsage,
+)
+from ray.data._internal.execution.util import make_ref_bundles
+from ray.data.tests.conftest import *  # noqa
+
+EMPTY_DOWNSTREAM_USAGE = collections.defaultdict(lambda: DownstreamMemoryInfo(0, 0))
+NO_USAGE = TopologyResourceUsage(ExecutionResources(), EMPTY_DOWNSTREAM_USAGE)
+
+
+@ray.remote
+def sleep():
+    time.sleep(999)
+
+
+def make_map_transformer(block_fn):
+    def map_fn(block_iter):
+        for block in block_iter:
+            yield block_fn(block)
+
+    return create_map_transformer_from_block_fn(map_fn)
+
+
+def make_ref_bundle(x):
+    return make_ref_bundles([[x]])[0]
```

Comment:
not a huge deal, should we import these from `test_streaming_executor.py` instead of redefining in this file?

Reply:
I don't think test file can import each other since these test files are not part of the ray package. Probably one solution is to move them into some test utils. I'll probably leave it to data team to do the refactoring if needed then ;).